backend/editor/entries.py

```diff
@@ -483,29 +493,27 @@ def update_node_children(self, entry, new_children_ids):
                 WHERE parent.id = $id AND deleted_child.id = $child
                 DELETE rel
             """
-            get_current_transaction().run(query, {"id": entry, "child": child})
+            await get_current_transaction().run(query, {"id": entry, "child": child})
 
         # Create non-existing nodes
         query = f"""
             MATCH (child:{self.project_name}:ENTRY)
                 WHERE child.id in $ids RETURN child.id
         """
-        existing_ids = [
-            record["child.id"]
-            for record in get_current_transaction().run(query, ids=list(added_children))
-        ]
+        _result = await get_current_transaction().run(query, ids=list(added_children))
+        existing_ids = [record["child.id"] for record in (await _result.data())]
         to_create = added_children - set(existing_ids)
 
         # Normalising new children node ID
         created_child_ids = []
 
         for child in to_create:
             main_language_code = child.split(":", 1)[0]
-            created_node_id = self.create_node("ENTRY", child, main_language_code)
+            created_node_id = await self.create_node("ENTRY", child, main_language_code)
             created_child_ids.append(created_node_id)
 
             # TODO: We would prefer to add the node just after its parent entry
-            self.add_node_to_end("ENTRY", created_node_id)
+            await self.add_node_to_end("ENTRY", created_node_id)
 
         # Stores result of last query executed
```

Comment:
I think we forget to create relations if the nodes are already existing. So, the function from this line, needs to be changed as follows:
```
        children_to_create = created_child_ids + existing_ids
        # Stores result of last query executed
        result = []
        for child in children_to_create:
            # Create new relationships if it doesn't exist
            query = f"""
                MATCH (parent:{self.project_name}:ENTRY), (new_child:{self.project_name}:ENTRY)
                WHERE parent.id = $id AND new_child.id = $child
                MERGE (new_child)-[r:is_child_of]->(parent)
            """
            _result = await get_current_transaction().run(query, {"id": entry, "child": child})
            result = list(await _result.value())

        return result
```

Reply:
Cool we also fix bugs while reviewing :-)